src/cpp/main.cpp

```diff
@@ -514,3 +529,17 @@ long getMicroseconds(){
 		return 0;
 	#endif
 }
+
+// http://burtleburtle.net/bob/hash/doobs.html
+unsigned long generateSeed(unsigned long a, unsigned long b, unsigned long c){
+	a=a-b; a=a-c; a=a^(c >> 13);
+	b=b-c; b=b-a; b=b^(a << 8);
+	c=c-a; c=c-b; c=c^(b >> 13);
+	a=a-b; a=a-c; a=a^(c >> 12);
+	b=b-c; b=b-a; b=b^(a << 16);
+	c=c-a; c=c-b; c=c^(b >> 5);
+	a=a-b; a=a-c; a=a^(c >> 3);
+	b=b-c; b=b-a; b=b^(a << 10);
+	c=c-a; c=c-b; c=c^(b >> 15);
+	return c;
+}
```

Comment:
This function does not appear to be licensed in a way that is compatible with the GPLv2.  This patch cannot be accepted with this function included.

Reply:
"You may use this code any way you wish, private, educational, or commercial.  It's free." - this code is public domain and compatible with GPLv2. Not sure you need that complex seed generator though